sympy/sets/fancysets.py

```diff
@@ -483,6 +483,128 @@ def _intersect(self, other):
             else:
                 return
 
+    def _union(self, other):
+        """
+
+        This function should only be used internally.
+        See Set._union for docstring
+
+        Used within the :class:`Union` class
+
+        This function club imageset `other` with `self` if there is difference
+        of `pi` or `2*pi` and returns this new imageset otherwise None.
+
+        Returns
+        =======
+        self._union(other) returns a new, joined set if self knows how
+        to join itself with other, otherwise it returns ``None``.
+
+        Examples
+        ========
+
+        >>> from sympy import Lambda
+        >>> from sympy.sets.fancysets import ImageSet
+        >>> from sympy import pprint, Union
+        >>> from sympy.core import Dummy, pi, S, Symbol
+        >>> n = Dummy('n')
+        >>> img1 = ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Integers)
+        >>> img2 = ImageSet(Lambda(n, 2*n*pi + 7*pi/4), S.Integers)
+        >>> pprint(img1._union(img2), use_unicode=False)
+                3*pi
+        {n*pi + ---- | n in Integers()}
+                 4
+
+        """
+
+        from sympy.core import pi
+        from sympy.polys import Poly
+        from sympy.simplify import simplify
+        from sympy.polys.polyerrors import PolynomialError
+
+        if other is S.EmptySet:
+            return self
+
+        def first_val(imgset):
+            base = imgset.base_set
+            if base == S.Reals:
+                return imgset.lamda(0)
+            try:
+                iterable = iter(base)
+                return imgset.lamda(next(iterable))
+            except TypeError:
+                raise TypeError
+
+        def _attempt_1(self_expr, final_expr, var_self, var_final, base):
+            var_term = self_expr.coeff(var_self)
+            if var_term == 0:
```

Comment:
It would make more sense if you can merge this clause with this [check](https://github.com/sympy/sympy/pull/11188/files#diff-c1c17bdd28aafd400bf9415c7d048382R589)